Content.Server/_CP14/Workbench/CP14WorkbenchSystem.UI.cs

```diff
@@ -4,14 +4,29 @@
  */
 
 using Content.Shared._CP14.Workbench;
+using Content.Shared._CP14.Workbench.Prototypes;
+using Robust.Shared.Prototypes;
 
 namespace Content.Server._CP14.Workbench;
 
 public sealed partial class CP14WorkbenchSystem
 {
+    [Dependency] private readonly SharedCP14WorkbenchSystem _sharedWorkbenchSystem = default!;
     private void OnCraft(Entity<CP14WorkbenchComponent> entity, ref CP14WorkbenchUiCraftMessage args)
     {
-        if (!entity.Comp.Recipes.Contains(args.Recipe))
+        var recipes = new List<ProtoId<CP14WorkbenchRecipePrototype>>();
+
+        foreach(var recipe in entity.Comp.Recipes)
```

Comment:
I don't like something about the optimization here, but I'll take a closer look later on